lib/spack/spack/schema/config.py

```diff
@@ -87,3 +100,39 @@
     'additionalProperties': False,
     'properties': properties,
 }
+
+
+def update(data):
+    """Update the data in place to remove deprecated properties.
+
+    Args:
+        data (dict): dictionary to be updated
+
+    Returns:
+        True if data was changed, False otherwise
+    """
+    changed = False
+    # currently only deprecated property is `install_tree`
+    install_tree = data.get('install_tree', None)
+    if isinstance(install_tree, six.string_types):
+        # deprecated short-form install tree
+        # add value as `root` in updated install_tree
+        data['install_tree'] = {'root': install_tree}
+        changed = True
+
+    install_scheme = data.pop('install_scheme', None)
```

Comment:
Should this be `install_path_scheme` (vs. `install_scheme`)? If so, I think you'll want to update the test you added too.